src/pages/index.js

```diff
@@ -2,46 +2,84 @@
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
+import { validatorConfig, cardsListSelector, apiData } from "../utils/constants.js";
 import "../pages/index.css";
-import { initialCards, validatorConfig, cardsListSection } from "../utils/constants.js";
 
 //select elements
 const profile = document.querySelector(".profile");
 const profileEditButton = profile.querySelector(".profile__edit-button");
 const profileAddButton = profile.querySelector(".profile__add-button");
-const profileName = profile.querySelector(".profile__title");
-const profileJob = profile.querySelector(".profile__description");
+const profileNameEl = profile.querySelector(".profile__title");
+const profileAboutEl = profile.querySelector(".profile__description");
+const profileAvatar = profile.querySelector(".profile__image");
+const profileAvatarEdit = profile.querySelector(".profile__image-edit");
 const profileModalForm = document.forms["profile-form"];
 const profileModalNameInput = profileModalForm.querySelector(".profile-modal__name-input");
 const profileModalDescInput = profileModalForm.querySelector(".profile-modal__desc-input");
 const addModalForm = document.forms["add-form"];
+const avatarEditForm = document.forms["avatar-edit-form"];
 
 //instantiate classes
-const userInfo = new UserInfo(profileName, profileJob);
+const userInfo = new UserInfo({
+  name: profileNameEl,
+  about: profileAboutEl,
+  avatar: profileAvatar,
+  apiHeaders: apiData.headers,
+  apiUser: apiData.currentUser,
+});
+userInfo.getUserInfo();
+
+const initialCardsApi = new Api({
+  url: apiData.cards,
+  headers: apiData.headers,
+});
+initialCardsApi.handleFetch().then((res) => {
+  cardsListSection.renderItems(res, true);
+});
+
+const cardsListSection = new Section((item, firstRender) => {
+  cardsListSection.addItem(createCard(item), firstRender);
+}, cardsListSelector);
+
 const profileFormValidator = new FormValidator(validatorConfig, profileModalForm);
 const addFormValidator = new FormValidator(validatorConfig, addModalForm);
+const avatarEditFormValidator = new FormValidator(validatorConfig, avatarEditForm);
 profileFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarEditFormValidator.enableValidation();
 
 const profilePopup = new PopupWithForm(
   "#profile-modal",
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    userInfo.setUserInfo(inputFieldValues);
+    userInfo.setUserInfo(inputFieldValues, profilePopup);
   },
   profileFormValidator
 );
 profilePopup.setEventListeners();
 
+const avatarEditPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  (inputValue, evt) => {
+    evt.preventDefault();
+    profileAvatar.src = inputValue.Link;
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change